Sources/TAPS/TCPServerService.swift

```diff
@@ -0,0 +1,68 @@
+// TCPServerService.swift
+// RFC-compliant TCP server service implementation
+
+import Foundation
+
+/// TCP server service implementing ServerServiceProtocol
+@available(macOS 15.0, *)
+public actor TCPServerService: ServerServiceProtocol, ServiceWithDefaults {
+    public typealias Parameters = TCPServerParameters
+    public typealias Client = TCPClient
+    
+    private let port: Int
+    
+    public init(port: Int) {
+        self.port = port
+    }
+    
+    /// Accept clients using withServer pattern with structured concurrency
+    public func withServer<T: Sendable>(
+        context: TAPSContext,
+        parameters: Parameters,
+        acceptClient: @escaping @Sendable (Client) async throws -> T
+    ) async throws -> T {
+        return try await TCPServer.withServer(
+            port: port,
+            parameters: parameters,
+            acceptClient: acceptClient
+        )
+    }
+    
+    // MARK: - ServiceWithDefaults
+    public static var defaultParameters: TCPServerParameters {
+        return TCPServerParameters.defaultParameters
+    }
+}
+
+extension ServerServiceProtocol where Self == TCPServerService {
+    public static func tcp(port: Int) -> TCPServerService {
```

Comment:
We'll need to pass in a hostname, but we can give it a default value like `"0.0.0.0"`